drake/systems/plants/collision/FCLModel.cpp

```
@@ -0,0 +1,421 @@
+  fcl::BVHModel<OBBRSS>* bvh_shape = new fcl::BVHModel<OBBRSS>();
+
+  // fcl does not have a method to generate a capsule mesh.
+  // TODO: Make one.
+  // fcl::generateBVHModel(*bvh_shape, shape, Transform3f(), numTheta,
+  // numLength);
+
+  return bvh_shape;
+}
+
+// TODO: Figure out how to convert mesh verts to fcl mesh.
+BVHModel<OBBRSS>* FCLModel::newFCLMeshShape(const DrakeShapes::Mesh& geometry,
+                                            bool use_margins) {
+  Matrix3Xd vertices;
+  if (geometry.extractMeshVertices(vertices)) {
+    // geometry.getPoints(Eigen::Matrix3Xd &points) const;
+  }
+  // What about traingle indexes?
+  // TODL........
+  cerr << "Warning: FCLModel::newMeshShape is not implemented." << endl;
+  return 0;
+}
+
+ElementId FCLModel::addElement(const Element& element) {
+  ElementId id = Model::addElement(element);
+
+  if (id != 0) {
+    BVHModel<OBBRSS>* bvh_shape = 0;
+    switch (elements[id]->getShape()) {
+      case DrakeShapes::BOX: {
+        const auto box =
+            static_cast<const DrakeShapes::Box&>(elements[id]->getGeometry());
+        bvh_shape = newFCLBoxShape(box, true);
+      } break;
+      case DrakeShapes::SPHERE: {
+        const auto sphere = static_cast<const DrakeShapes::Sphere&>(
+            elements[id]->getGeometry());
+        bvh_shape = newFCLSphereShape(sphere, true);
+      } break;
+      case DrakeShapes::CYLINDER: {
+        const auto cylinder = static_cast<const DrakeShapes::Cylinder&>(
+            elements[id]->getGeometry());
+        bvh_shape = newFCLCylinderShape(cylinder, true);
+      } break;
+      case DrakeShapes::MESH: {
+        const auto mesh =
+            static_cast<const DrakeShapes::Mesh&>(elements[id]->getGeometry());
+        // I only see points.  I cannot find triangles.
+        // bt_shape = newBulletMeshShape(mesh, true);
+        // bt_shape_no_margin = newBulletMeshShape(mesh, false);
+        cerr << "Warning: FCLMesh not implemented." << endl;
```

Maybe throw an exception here? the code won't run without this implemented